replica/table.cc

```diff
@@ -898,7 +899,7 @@ utils::chunked_vector<compaction_group*> tablet_storage_group_manager::compactio
     size_t candidate_end = tr.end() ? storage_group_id_for_token(tr.end()->value()) : (storage_groups().size() - 1);
 
     while (candidate_start <= candidate_end) {
-        auto& sg = storage_groups()[candidate_start++];
+        auto& sg = storage_groups().at(candidate_start++);
```

Comment:
This won't work with dynamic allocation of storage groups
since `at()` would throw if the sg isn't populated.